Parsers/ASimRegistryEvent/Parsers/vimRegistryEventTrendMicroVisionOne.yaml

```diff
@@ -0,0 +1,137 @@
+  Link: https:/aka.ms/AboutASIM
+- Title: Trend Micro Vision One documentation
+  Link: 
+    https://docs.trendmicro.com/en-us/enterprise/trend-vision-one/xdr-part/search-app/data-mapping-intro/data-mapping-detecti.aspx
+    https://automation.trendmicro.com/xdr/api-v3#tag/Observed-Attack-Techniques-Pipeline/paths/~1v3.0~1oat~1dataPipelines~1%7Bid%7D~1packages~1%7BpackageId%7D/get
+    https://automation.trendmicro.com/xdr/api-v3#tag/Observed-Attack-Techniques/paths/~1v3.0~1oat~1detections/get
+Description: |
+  This ASIM parser supports normalizing Trend Micro Vision One logs to the ASIM Registry Event normalized schema. Trend Micro Vision One events are captured through Trend Vision One data connector which ingests XDR logs into Microsoft Sentinel through the Trend Vision One API.
+ParserName: vimRegistryEventTrendMicroVisionOne
+EquivalentBuiltInParser: _Im_RegistryEvent_TrendMicroVisionOne
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventTypeLookup = datatable(detail_eventSubId_s: string, EventType: string)[
+    "TELEMETRY_REGISTRY_CREATE", "RegistryKeyCreated",
+    "TELEMETRY_REGISTRY_SET", "RegistryValueSet",
+    "TELEMETRY_REGISTRY_DELETE", "RegistryKeyDeleted",
+    "TELEMETRY_REGISTRY_RENAME", "RegistryKeyRenamed"
+  ];
+  let RegistryKeyPrefixLookup = datatable(
+      RegistryKeyPrefix: string,
+      RegistryKeyNormalizedPrefix: string
+  )[
+      "HKLM", "HKEY_LOCAL_MACHINE",
+      "HKU", "HKEY_USERS",
+      "HKCU", "HKEY_CURRENT_USER",
+      "HKCR", "HKEY_CLASSES_ROOT",
+      "HKCC", "HKEY_CURRENT_CONFIG"
+  ];
+  let RegistryValueTypeLookup = datatable (detail_objectRegType_d: real, RegistryValueType: string)[
+    0, "Reg_None",
+    1, "Reg_Sz",
+    2, "Reg_Expand_Sz",
+    3, "Reg_Binary",
+    4, "Reg_DWord",
+    5, "Reg_DWord",
+    7, "Reg_Multi_Sz",
+    11, "Reg_QWord"
+  ];
+  let EventSeverityLookup = datatable(detail_filterRiskLevel_s: string, EventSeverity: string)[
+      "low", "Low",
+      "medium", "Medium",
+      "high", "High",
+      "info", "Informational",
+      "critical", "High"
+  ];
+  let parser = (disabled: bool=false) {
+      TrendMicro_XDR_OAT_CL
+      | where not(disabled)
+      | where detail_eventId_s == "TELEMETRY_REGISTRY"
+      | parse filters_s with * "[" filters: string "]"
+      | parse-kv filters as (description: string, name: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
+      | lookup EventTypeLookup on detail_eventSubId_s
+      | lookup RegistryValueTypeLookup on detail_objectRegType_d
+      | lookup EventSeverityLookup on detail_filterRiskLevel_s
+      | invoke _ASIM_ResolveDvcFQDN('detail_endpointHostName_s')
+      | extend RegistryKeyPrefix = tostring(split(detail_objectRegistryKeyHandle_s, @'\')[0])
+      | lookup RegistryKeyPrefixLookup on RegistryKeyPrefix
+      | extend 
+          RegistryKey = replace_string(detail_objectRegistryKeyHandle_s, RegistryKeyPrefix, RegistryKeyNormalizedPrefix),
```

Comment:
In the attached sample logs, EventType is 'RegistryValueSet' but no value exists in Registry fields.
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/83f923dc-c1c8-43ce-b9d6-8e38934ccf50)